src/components/controller/Gamepad.vue

```diff
@@ -0,0 +1,62 @@
+<!-- Logical Component so can take advantage of vue lifecycle functions like onMount -->
+<script setup lang="ts">
+import { onMounted, onUnmounted } from 'vue';
+import { ControllerState } from '../../lib/controller/controllerState';
+import { useControllerStore } from '@/store/useController';
+
+const DELTA_SENSITIVITY = 0.01;
+const POLLING_RATE_IN_HERTZ = 20;
+
+const indexToControllerName = new Map();
+const indexToControllerState = new Map();
+// const isGamepadConnected = ref<boolean>(false);
+const controller = useControllerStore();
+
+onMounted(() => {
+  window.addEventListener('gamepadconnected', onGamePadConnectsHandler);
+  window.addEventListener('gamepaddisconnected', onGamePadDisconnectsHandler);
+});
+onUnmounted(() => {
+  window.removeEventListener('gamepadconnected', onGamePadConnectsHandler);
+  window.removeEventListener('gamepadconnected', onGamePadDisconnectsHandler);
+});
+
+function onGamePadConnectsHandler(e: GamepadEvent) {
+  console.log('Controller connected with index %d\n' + e.gamepad.id, e.gamepad.index);
+  indexToControllerName.set(e.gamepad.index, e.gamepad.id);
+  const state = new ControllerState(
+    'src\\lib\\controller\\drivebaseAndArmBinding.json',
```

Comment:
This isn't necessarily a problem with this PR, just something that caught my eye and might be worth discussing. Do you know why we're using backslashes here? I need to test it, but I wonder if it might cause incompatibilities between platforms.

Reply:
it's because they're using fetch to get the json file, which you ideally shouldn't be doing. vite has built in functionality to import json files: https://vite.dev/guide/features#json. you can pass them into controller state as an object or import them dynamically using glob import: https://vite.dev/guide/features#glob-import

Reply:
Gotcha, I'll address this in my controller changes then.

Reply:
Thanks for mentioning this Kyler. Yeah please get rid of fetch, and yeah just import it directly. I wouldn't use glob as the json files are static and not going to change, but yeah in the future will do this.

Reply:
Yeah I think this is where we cannot fully don't know what the driver(s) want(s). Yes, there will be different control configurations like tank controls and etc. Hm yeah I guess its the way you want to code it up as the benefit of static is not worrying about async stuff. Thinking of it now, glob import makes the most sense as it actually would be very useful if it is dynamic. A cool feature would have each json file having metadata key-values (aka like 2 keys at the top) that specifies the title like "tank control" & "Bob's Standard Control" and perhaps color.  The title and color would then be used in the navbar bar to display what control mode it is in. I know Vlad mentioned that the driver should be able to change the controls while the code is running, but I think that is very low priority as that would involve either storing it on local storage (then another computer wouldn't work) or automatically saving the json which the amount of time and effort isn't worth it. I think the easiest thing is if the driver really wants their own controls, they can copy the json file and change the values themselves, which isn't very hard and they shouldn't need to do it a lot.